### PR TITLE
refactor: command

### PR DESCRIPTION
**Pull request by irainia**
This PR is to address #264 .

This PR is to refactor the currently existing `cmd` package. The following is the general changes in this PR:

* Each main command is put under a dedicated package. For example `job` package under `cmd` as `job` here is considered as the main command. Main command is the sub-command that is directly added by the root, mostly available under `cmd/commands.go`.
* Each sub-command of the main command is put in the same package. For example, `job create` is put under `job` package, but different file.
* Each main command and its sub-command (if exist) uses a dedicated structure if that command involves execution. For example, in `job` it has structure `jobCommand` with its own functionalities. But, if we check `admin build` sub command, there's no such structure as there's no business execution specific to that sub-command.
* Some refactors on some implementations, including renaming for better clarity or to simplification, extraction function, and more.
* Some bug fixes that was previously missed, such as namespace describe and listing secret.

### CODE REVIEW

File: cmd/admin/build_instance.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		ProjectName:  b.clientConfig.Project.Name,
+		JobName:      jobName,
+		ScheduledAt:  jobScheduledTimeProto,
+		InstanceType: pb.InstanceSpec_Type(pb.InstanceSpec_Type_value[utils.ToEnumProto(b.runType, "type")]),
+		InstanceName: b.runName,
+	}
+	return jobRun.RegisterInstance(conn.GetContext(), request)
+}
+
+func (b *buildInstanceCommand) getJobScheduledTimeProto() (*timestamppb.Timestamp, error) {
```
Comment (sbchaos) [inline cmd/admin/build_instance.go:197]: We are not changing the struct, we should not be using pointer unless struct is big.

File: cmd/admin/build_instance.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	for fileName, fileContent := range jobResponse.Context.Files {
+		filePath := filepath.Join(dirPath, fileName)
+		if err := writeToFileFn(filePath, fileContent, b.logger.Writer()); err != nil {
+			return fmt.Errorf("failed to write asset file at %s: %w", filePath, err)
+		}
+	}
+	return nil
+}
+
+func (b *buildInstanceCommand) sendInstanceRequest(jobName string, jobScheduledTimeProto *timestamppb.Timestamp) (*pb.RegisterInstanceResponse, error) {
```
Comment (sbchaos) [inline cmd/admin/build_instance.go:178]: Is the pointer required here?

File: cmd/admin/build_instance.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	if err := writeToFileFn(filePath, envFileBlob, b.logger.Writer()); err != nil {
+		return fmt.Errorf("failed to write asset file at %s: %w", filePath, err)
+	}
+	return nil
+}
+
+func (b *buildInstanceCommand) writeJobResponseMapToFiles(
+	jobResponse *pb.RegisterInstanceResponse, dirPath string,
+) error {
+	if err := os.MkdirAll(dirPath, 0o777); err != nil {
```
Comment (sbchaos) [inline cmd/admin/build_instance.go:164]: The file permission of `0o777` is too open, can we restrict it to `600` or similar

File: cmd/admin/build_instance.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+	filePath := filepath.Join(dirPath, models.InstanceDataTypeEnvFileName)
+	writeToFileFn := utils.WriteStringToFileIndexed()
+	if err := writeToFileFn(filePath, envFileBlob, b.logger.Writer()); err != nil {
+		return fmt.Errorf("failed to write asset file at %s: %w", filePath, err)
+	}
+	return nil
+}
+
+func (b *buildInstanceCommand) writeJobResponseMapToFiles(
```
Comment (sbchaos) [inline cmd/admin/build_instance.go:161]: Can we rename the function to `writeJobAssetsToFiles`, current name is not very descriptive.

File: cmd/backup/create.go [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+		if errors.Is(err, context.DeadlineExceeded) {
+			c.logger.Error(logger.ColoredError("Backup took too long, timing out"))
+		}
+		return fmt.Errorf("request failed to backup job %s: %w", backupRequest.ResourceName, err)
+	}
+	c.printBackupResponse(backupResponse)
+	return nil
+}
+
+func (c *createCommand) printBackupResponse(backupResponse *pb.CreateBackupResponse) {
```
Comment (sbchaos) [inline cmd/backup/create.go:135]: Can we use non-pointer here

File: cmd/backup/create.go [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+	c.logger.Info(logger.ColoredSuccess("Resource backup completed successfully:"))
+	for counter, result := range backupResponse.Urn {
+		c.logger.Info(fmt.Sprintf("%d. %s", counter+1, result))
+	}
+}
+
+func (c *createCommand) getAllowedDownstreamNamespaces(namespaceName string) []string {
```
Comment (sbchaos) [inline cmd/backup/create.go:142]: Simlify to a function, pass allDownstream as param.
```go
func getAllowedDownstreamNamespaces(namespaceName string, allDownstream bool) []string {
  if allDownstream {
      return []string{"*"}
  }
  return []string{namespaceName}
}
```

File: cmd/version/version.go [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+	v.printAllPluginInfos()
+	return nil
+}
+
+func (v *versionCommand) PostRunE(_ *cobra.Command, _ []string) error {
+	v.pluginCleanFn()
+	return nil
+}
+
+func (v *versionCommand) printAllPluginInfos() {
```
Comment (sbchaos) [inline cmd/version/version.go:94]: Check pointer semantic usage

File: cmd/replay/create.go [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+
+	replayID, err := c.runReplayRequest(jobName, startDate, endDate)
+	if err != nil {
+		return err
+	}
+	c.logger.Info(logger.ColoredSuccess("Replay request created with id %s", replayID))
+	return nil
+}
+
+func (c *createCommand) getAllowedDownstreamNamespaces() []string {
```
Comment (sbchaos) [inline cmd/replay/create.go:124]: Convert to function

File: cmd/replay/list.go [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+		"Start Date",
+		"End Date",
+		"Ignore Downstream?",
+		"Requested At",
+		"Status",
+	})
+
+	for _, replaySpec := range replayListResponse.ReplayList {
+		table.Append([]string{
+			replaySpec.Id, replaySpec.JobName, replaySpec.StartDate.AsTime().Format(models.JobDatetimeLayout),
```
Comment (sbchaos) [inline cmd/replay/list.go:96]: Please put single item on one line.

File: cmd/admin/build_instance.go (hunk 3 above)
Comment (irainia) [reply cmd/admin/build_instance.go]: updated

File: cmd/admin/build_instance.go (hunk 4 above)
Comment (irainia) [reply cmd/admin/build_instance.go]: updated

File: cmd/backup/create.go (hunk 6 above)
Comment (irainia) [reply cmd/backup/create.go]: updated

File: cmd/replay/create.go (hunk 8 above)
Comment (irainia) [reply cmd/replay/create.go:125]: updated

File: cmd/replay/list.go (hunk 9 above)
Comment (irainia) [reply cmd/replay/list.go]: updated

File: cmd/initialize/initialize.go [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+func (i *initializeCommand) initClientConfig(clientConfig *config.ClientConfig) error {
+	if err := i.setupDirPathForClientConfig(clientConfig); err != nil {
+		return err
+	}
+	marshalledClientConfig, err := yaml.Marshal(clientConfig)
+	if err != nil {
+		return err
+	}
+	filePath := i.getClientConfigPath()
+	filePermission := 0o660
```
Comment (sbchaos) [inline cmd/initialize/initialize.go:85]: Do we need permission to be `0o660` ? Can we restrict it to `0o600` ?

**Review by sbchaos — APPROVED** [review]

**Pull request merged**